### PR TITLE
[BEAM-3240] Improves development and testing instructions related to Python SDK

### PR DESCRIPTION
**Pull request by chamikaramj**
Updates contribution guide to include development and testing instructions for Python SDK.

### CODE REVIEW

**Comment by chamikaramj** [comment]
R: @aaltay @davorbonaci

File: src/contribute/contribution-guide.md [hunk 1]
```diff
@@ -231,10 +235,54 @@ Then you can push your local, committed changes to your (forked) repository on G
...
+
 For contributions to the Java code, run unit tests locally via Maven. Alternatively, you can use Travis-CI.
 
     $ mvn clean verify
 
+#### Python SDK
+
+For contributions to the Python code, you can use following commands to run unit tests locally and to check for lint errors. Once you send a pull request for review, these tests will be rerun using Jenkins. We recommend setting up a virtual environment before testing your code.
+
+##### Running a single test using nose
```
Comment (aaltay) [inline src/contribute/contribution-guide.md]: Should we add a section for testing with tox? (not just for lint)

File: src/contribute/contribution-guide.md [hunk 2]
```diff
@@ -231,10 +235,54 @@ Then you can push your local, committed changes to your (forked) repository on G
...
 
+#### Java SDK
+
 For contributions to the Java code, run unit tests locally via Maven. Alternatively, you can use Travis-CI.
 
     $ mvn clean verify
 
+#### Python SDK
+
+For contributions to the Python code, you can use following commands to run unit tests locally and to check for lint errors. Once you send a pull request for review, these tests will be rerun using Jenkins. We recommend setting up a virtual environment before testing your code.
```
Comment (aaltay) [inline src/contribute/contribution-guide.md]: Maybe a TBD for people who wants to use `mvn`. FWIW `mvn clean verify` works in the python directory. It should work from the root with the right selection of project (`mvn clean verify -pl sdks/python`)

File: src/contribute/contribution-guide.md [hunk 3]
```diff
@@ -231,10 +235,54 @@ Then you can push your local, committed changes to your (forked) repository on G
 ### Testing
 All code should have appropriate unit testing coverage. New code should have new tests in the same contribution. Bug fixes should include a regression test to prevent the issue from reoccurring.
 
+#### Java SDK
+
 For contributions to the Java code, run unit tests locally via Maven. Alternatively, you can use Travis-CI.
 
     $ mvn clean verify
 
+#### Python SDK
```
Comment (aaltay) [inline src/contribute/contribution-guide.md]: Should we say anything about cython? (probably not)

File: src/contribute/contribution-guide.md (hunk 2 above)
Comment (chamikaramj) [inline src/contribute/contribution-guide.md]: I think we should not add that info till https://issues.apache.org/jira/browse/BEAM-2341 and other issues around Maven+Python SDK are resolved.

File: src/contribute/contribution-guide.md (hunk 1 above)
Comment (chamikaramj) [inline src/contribute/contribution-guide.md]: Why is it needed ? If it simply another way to run tests I would rather avoid adding that info here.

File: src/contribute/contribution-guide.md (hunk 2 above)
Comment (aaltay) [inline src/contribute/contribution-guide.md]: I wonder if adding some information would help people coming from the mvn background. I am not sure, we can leave it as is.

**Comment by aaltay** [comment]
LGTM. Thank you. (please self merge)

File: src/contribute/contribution-guide.md (hunk 2 above)
Comment (chamikaramj) [inline src/contribute/contribution-guide.md]: Added instructions for running tests using Maven.

Comment (davorbonaci) [inline src/contribute/contribution-guide.md]: Don't follow -- which are "following commands"?

Fix above: Travis CI is disabled.
Overall, I'd probably generalize and say that mvn clean verify does it all, i.e., no Java SDK/Python SDK sections at all.

**Comment by chamikaramj** [comment]
Did following updates.

* Removed references to travis.
* Removed references to "nose" since a single test can be run without that (added this command).

Given that Maven support for Python SDK is limited (does not work from a virtual environment, cannot be used to rune a single test, etc.) I think having these instructions will be useful for potential contributors to Python SDK.

**Comment by chamikaramj** [comment]
Regarding "I wonder should all things under ### Testing go to the testing page" I think it's better to keep instructions related to testing that applies to most PRs in the contributors guide (information I added) while keeping any additional information (for example Maven commands that users sometimes need) in a separate document. Davor, could you PTAL.

File: src/contribute/contribution-guide.md [hunk 4]
```diff
@@ -98,6 +96,10 @@ Add your forked repository as an additional Git remote, where you’ll push your
 
 You are now ready to start developing!
 
+#### [Python SDK Only] Setup a virtual environemt
```
Comment (davorbonaci) [inline src/contribute/contribution-guide.md]: Setup -> Set up?

File: src/contribute/contribution-guide.md [hunk 5]
```diff
@@ -98,6 +96,10 @@ Add your forked repository as an additional Git remote, where you’ll push your
...
+
+We recommend setting up a virtual envioment for developing Python SDK. Please see instructions available at [Quickstart (Python)]({{ site.baseurl }}/get-started/quickstart-py/) for setting up a virtual environment.
```
Comment (davorbonaci) [inline src/contribute/contribution-guide.md]: at quickstart -> in quickstart?

File: src/contribute/contribution-guide.md [hunk 6]
```diff
@@ -231,10 +233,31 @@ Then you can push your local, committed changes to your (forked) repository on G
...
-For contributions to the Java code, run unit tests locally via Maven. Alternatively, you can use Travis-CI.
+#### Java SDK
+
+For contributions to the Java code, run unit tests locally via Maven.
 
     $ mvn clean verify
 
+#### Python SDK
+
+For contributions to the Python code, you can use command given below to run unit tests locally. If you update any of the [cythonized](http://cython.org) files in python SDK, you must install "cython" package before running following command to properly test your code. We recommend setting up a virtual environment before testing your code.
```
Comment (davorbonaci) [inline src/contribute/contribution-guide.md]: python -> Python

File: src/contribute/contribution-guide.md (hunk 4 above)
Comment (chamikaramj) [inline src/contribute/contribution-guide.md]: Done.

File: src/contribute/contribution-guide.md (hunk 5 above)
Comment (chamikaramj) [inline src/contribute/contribution-guide.md]: Done.

File: src/contribute/contribution-guide.md (hunk 6 above)
Comment (chamikaramj) [inline src/contribute/contribution-guide.md]: Done.

**Comment by aaltay** [comment]
LGTM. Thank you, please self merge.

**Comment by chamikaramj** [comment]
Thanks.

**Pull request merged**